modules/rosa-sts-cluster-terraform-customization.adoc

```diff
@@ -0,0 +1,754 @@
+    https_proxy             = string           # required  https proxy
+    additional_trust_bundle = optional(string) # a string contains contains a PEM-encoded X.509 certificate bundle that will be added to the nodes' trusted certificate store.
+    no_proxy                = optional(string) # no proxy
+  })
+}
+
+#Private link cluster info
+variable "private_cluster" {
+  type        = bool
+  description = "Do you want this cluster to be private? True or False."
+  default     = false
+}
+
+#VPC Info
+variable "create_vpc" {
+  default     = false
+  type        = bool
+  description = "Would you like to make a new VPC for your ROSA cluster? true or false"
+}
+
+variable "vpc_name" {
+  type        = string
+  description = "VPC Name"
+  default     = "rosa-tf-vpc"
+}
+
+variable "vpc_cidr_block" {
+  type        = string
+  description = "The value of the IP address block for machines or cluster nodes for the VPC."
+  default     = "10.0.0.0/16"
+}
+
+variable "private_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the private subnets"
+}
+
+variable "public_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the public subnets"
+  default     = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+}
+
+variable "single_nat_gateway" {
+  type        = bool
+  description = "Single NAT or per NAT for subnet"
+  default     = false
+}
+
+#AWS Info
+variable "aws_region" {
+  type    = string
+  default = "us-east-1"
+}
+
+variable "private_subnet_ids" {
+  type        = list(any)
+  description = "VPC private subnets IDs for ROSA Cluster"
+  default     = []
+}
+EOF
+----
+
+.. Run the following command to create the `vpc.tf` file:
++
+[source,terminal]
+----
+$ cat<<-EOF>vpc.tf
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "5.1.2"
+
+  count = var.create_vpc ? 1 : 0
+  name  = var.vpc_name
+  cidr  = var.vpc_cidr_block
+
+  azs             = local.region_azs
+  private_subnets = var.private_subnet_cidrs
+  public_subnets  = var.public_subnet_cidrs
+
+  enable_nat_gateway   = true
+  single_nat_gateway   = var.single_nat_gateway
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  tags = var.additional_tags
+}
+EOF
+----
+endif::tf-defaults[]
+. Create a `terraform.tfvars` file by running the following command:
++
+[source,terminal]
+----
+$ cat<<-EOF>terraform.tfvars
+# Advanced ROSA Cluster Options
+worker_node_replicas = "2"
+
+#Private Link Cluster & VPC
+single_nat_gateway=true
+ifndef::tf-defaults[]
+vpc_cidr_block="10.12.0.0/16"
+endif::tf-defaults[]
+private_subnet_cidrs=["10.12.0.0/22", "10.12.4.0/22", "10.12.8.0/22"]
+public_subnet_cidrs=["10.12.128.0/24", "10.12.129.0/24", "10.12.130.0/24"]
+EOF
+----
+
+. In your terminal, run the following command in the directory where you saved the Terraform files to set up Terraform to create these resources:
```

Comment:
```suggestion
. Start up a working directory containing the Terraform configuration files by running the following command:
```
Aren't we already in the terminal? And shouldn't we already be in the relevant directory whether we completed Step 1 or not (because the Terraform files go to the current directory no matter what, we said in Step 1).

If this is a common breaking step because the user doesn't do this process all at once, an Important admonition might be better; something like, "You must complete the following steps from the directory where you created the Terraform files."